compiler/src/dmd/declaration.d

```diff
@@ -1314,9 +1314,17 @@ extern (C++) class VarDeclaration : Declaration
 
     override final bool isImportedSymbol() const
     {
-        if (visibility.kind == Visibility.Kind.export_ && !_init && (storage_class & STC.static_ || parent.isModule()))
-            return true;
-        return false;
+        /* If global variable has `export` and `extern` then it is imported
+         *   export int sym1;            // definition:  exported
```

Comment:
Are you sure that is how it is supposed to work? This way you cannot import a D file from a library that is compiled into a DLL.

LDCs approach is to distinguish between root modules and non-root modules and it looks like the most feasible approach. I have started implementing that for dmd: https://github.com/dlang/dmd/pull/14849

Reply:
I was thinking of modifying .di generation. Currently,
```
int a = 5;
int b;
extern int c;
void foo() { }
void bar();
extern void abc();
```
results in:
```
// D import file generated from 'test4.d'
int a = 5;
int b;
extern int c;
void foo();
void bar();
extern void abc();
```
Note that `foo()` was turned from a declaration into a definition, but `a` and `b` were not. They should be. This should make .di header generation much better for DLLs.

Reply:
Ack, I meant "definition into declaration".

I appreciate your advice on this.

Reply:
Changing definitions into declarations doesn't help deciding whether it should be dll-exported or not, unless you want to do this explicitly and have some rules to propagate this to internal symbol names. I don't think adding `export` everywhere is desirable, consider annotating phobos. How can you still build a statically linked version of  the library?

Reply:
Indeed. What LDC does right now is usable, even if the language-level stuff isn't right. Let's stick to just implementing that for the time being and leave the QoL stuff until after we've fixed the language-level stuff.

It's mostly just that export needs to be redesigned, I am working on it. One of the reasons is due to [LNK4217](https://learn.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-warning-lnk4217?view=msvc-170) which I am indeed running into (warning, so can be ignored). But it has to be done at the DIP level to fix this. This is going to be a very tricky problem to solve.

Reply:
This is already implemented. No reason not to move forward with it.

Reply:
IIUC you are changing the rules for `export` (w/wo extern), so that global symbols are a bit more intuitive for hand crafted di files. Maybe it should have been this way from the start, but I doubt that this will be what is desired in the long run, i.e. use the same d file for compilation and import. For static variables the combination `export extern static` feels odd, too.

As this might break existing code that uses the current rules (w/wo initializer), an explicit changelog entry should be added.

Reply:
added changelog entry